core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -1195,7 +1195,7 @@ public Sha256Hash hashForSignature(int inputIndex, byte[] connectedScript, byte
             // EC math so we'll do it anyway.
             for (int i = 0; i < tx.inputs.size(); i++) {
                 TransactionInput input = tx.getInput(i);
-                input.clearScriptBytes();
+                input = input.withoutScriptBytes();
```

Comment:
These two "without" calls could be chained. I would even consider chaining the interior of the `for` to one statement.

Reply:
Very valid comment. I deliberately choose against chaining this (and others in this PR) because of the simplicity (reviewability) of the diff. I thought the chaining can be done later, maybe on a case-by-case basis. Or maybe we rewrite the entire Transaction(Builder) class anyway…